### DISCUSSION TITLE
Wokwi support added

### DISCUSSION THREAD

**Discussion by Vollbrecht** [Rahix/avr-hal#415] [Show and tell]
Hi guys,

I added support for Rust on AVR inside Wokwi.com. Wokwi is a free online simulator for different MCU's including some AVR chips. You can write the code live, compile it in the browser and run digital simulations with it alongside. From attaching a Logic Analyzer to letting a Stepper Motor run, it's all possible and simply editable. If you want to play around with it, you can find links to template project's and some examples here https://github.com/Vollbrecht/wokwi-rust-avr-projects. It's also possible to use the simulator inside VSCode with an extension. Here is an example of an [Arduino Uno Board with Rotary Encoder + Stepper Motor Running Rust Avr code](https://wokwi.com/projects/363948225373532161). The option to add a new project will be added later. You just can use one of the templates as a starting point. 

Thank you so much for your great work here. That made it extremely simple to get good integration into wokwi!

Hope you enjoy!

**Comment by Rahix** [comment]
(Converted this issue to a discussion.)

That's very cool, thanks for sharing! :)
